File: sdk-python/agentrix/resources/merchants.py
"""
Merchant resource for Agentrix SDK
"""

from typing import Dict, Any, Optional
from ..http_client import AgentrixClient


class MerchantResource:
    """Merchant operations"""

    def __init__(self, client: AgentrixClient):
        self.client = client

    def create_product(self, request: Dict[str, Any]) -> Dict[str, Any]:
        """Create a product"""
        if not request.get("name"):
            raise ValueError("Product name is required")
        if not request.get("price") or request["price"] <= 0:
            raise ValueError("Product price must be a positive number")
        return self.client.post("/products", request)

    def get_product(self, product_id: str) -> Dict[str, Any]:
        """Get product by ID"""
        if not product_id:
            raise ValueError("Product ID is required")
        return self.client.get(f"/products/{product_id}")

    def list_products(
        self,
        page: Optional[int] = None,
        limit: Optional[int] = None,
        category: Optional[str] = None,
        search: Optional[str] = None,
    ) -> Dict[str, Any]:
        """List products"""
        params = {}
        if page is not None:
            params["page"] = page
        if limit is not None:
            params["limit"] = limit
        if category:
            params["category"] = category
        if search:
            params["search"] = search

        return self.client.get("/products", params=params if params else None)

    def update_product(self, product_id: str, updates: Dict[str, Any]) -> Dict[str, Any]:
        """Update a product"""
        if not product_id:
            raise ValueError("Product ID is required")
        return self.client.put(f"/products/{product_id}", updates)

    def delete_product(self, product_id: str) -> None:
        """Delete a product"""
        if not product_id:
            raise ValueError("Product ID is required")
        self.client.delete(f"/products/{product_id}")

    def get_order(self, order_id: str) -> Dict[str, Any]:
        """Get order by ID"""
        if not order_id:
            raise ValueError("Order ID is required")
        return self.client.get(f"/orders/{order_id}")

    def list_orders(
        self,
        page: Optional[int] = None,
        limit: Optional[int] = None,
        status: Optional[str] = None,
    ) -> Dict[str, Any]:
        """List orders"""
        params = {}
        if page is not None:
            params["page"] = page
        if limit is not None:
            params["limit"] = limit
        if status:
            params["status"] = status

        return self.client.get("/orders", params=params if params else None)

File: sdk-js/src/resources/merchants.ts
/**
 * Merchant resource for Agentrix SDK
 */

import { AgentrixClient } from '../client';
import {
  Product,
  CreateProductRequest,
  Order,
} from '../types/merchant';

export class MerchantResource {
  constructor(private client: AgentrixClient) {}

  /**
   * Create a product
   * When availableToAgents is true, the product will be:
   * 1. Stored in database
   * 2. Generate embedding for title + description
   * 3. Indexed in vector database (for semantic search)
   * 4. Synced to Marketplace Catalog service
   * 5. Made available for AI Agent retrieval
   */
  async createProduct(request: CreateProductRequest): Promise<Product> {
    if (!request.name) {
      throw new Error('Product name is required');
    }
    if (!request.price || request.price <= 0) {
      throw new Error('Product price must be a positive number');
    }
    
    // Default availableToAgents to true if not specified
    const productData = {
      ...request,
      availableToAgents: request.availableToAgents !== false, // Default to true
    };
    
    return this.client.post<Product>('/products', productData);
  }

  /**
   * Get product by ID
   */
  async getProduct(id: string): Promise<Product> {
    if (!id) {
      throw new Error('Product ID is required');
    }
    return this.client.get<Product>(`/products/${id}`);
  }

  /**
   * List products
   */
  async listProducts(params?: {
    page?: number;
    limit?: number;
    category?: string;
    search?: string;
  }): Promise<{ data: Product[]; pagination: any }> {
    return this.client.get('/products', { params });
  }

  /**
   * Update a product
   */
  async updateProduct(id: string, updates: Partial<CreateProductRequest>): Promise<Product> {
    if (!id) {
      throw new Error('Product ID is required');
    }
    return this.client.put<Product>(`/products/${id}`, updates);
  }

  /**
   * Delete a product
   */
  async deleteProduct(id: string): Promise<void> {
    if (!id) {
      throw new Error('Product ID is required');
    }
    return this.client.delete(`/products/${id}`);
  }

  /**
   * Get order by ID
   */
  async getOrder(id: string): Promise<Order> {
    if (!id) {
      throw new Error('Order ID is required');
    }
    return this.client.get<Order>(`/orders/${id}`);
  }

  /**
   * List orders
   */
  async listOrders(params?: {
    page?: number;
    limit?: number;
    status?: string;
  }): Promise<{ data: Order[]; pagination: any }> {
    return this.client.get('/orders', { params });
  }
}

